### ISSUE TITLE
Cannot change user settings without email address

### ISSUE THREAD

**Issue by pafade89** [Fallenbagel/jellyseerr#1094]
### Description

Hi, 
when you want to change user specific settings for other users, without enter an email address you get the following error:

![image](https://github.com/user-attachments/assets/fe25f4ea-c78f-4ad2-9002-6e07fb8c6887)
![image](https://github.com/user-attachments/assets/320e6890-f5ad-4fbc-a2a9-0a42aa4eac10)

### Version

2.1.0

### Steps to Reproduce

1. Goto Users
2. Click edit user
3. Change as example the weekly limits 
4. Click safe

### Platform

desktop

### Device

PC

### Operating System

Windows 11

### Browser

Chrome

### Code of Conduct

- [X] I agree to follow Jellyseerr's Code of Conduct

**Comment by Fallenbagel** [comment]
This is not a bug. Email address is a must for every user. It should not be empty. It can be your jellyfin/emby username as well. This is whats used in local logins.

**Comment by pafade89** [comment]
Hi, after update to version 2.2.2 it is still not possible to change the count of request for a user without an email address.

![image](https://github.com/user-attachments/assets/3119bed1-5eac-4e41-be6e-eba924694916)
![image](https://github.com/user-attachments/assets/6e192b5f-a975-42c7-8152-7920fae4e9f7)

Please reopen. Thx in advance.

**Comment by Fallenbagel** [comment]
@pafade89 actually this is fixed. The reason yours is not working is due to the missing values in the db.
How the current flow works is, a user can have an empty email (but in reality in the db, the user has an "email" value with their jellyfin/emby username) 

(see the video below)

https://github.com/user-attachments/assets/3352c2f4-e591-4e81-ae25-827bd5fa85b1

in your case the issue is all these users already have an empty email value in the db. You can fix this by just adding a bogus email and then removing it (it will revert back to empty email with db having the username). 
Unfortunately, we do not have an auto-fix for that implemented with those dbs that were affected with the null values during the version where validation was removed. But what the PR and this issue wants to achieve, is already achieved in 2.2.x

**Comment by pafade89** [comment]
Try to change email via APIto nothing, but response from API is read only:

```
{
  "message": "request.body.email is read-only",
  "errors": [
    {
      "path": ".body.email",
      "message": "is read-only",
      "errorCode": "readOnly.openapi.validation"
    }
  ]
}
```

**Comment by gauthier-th** [comment]
I'm pretty confident this is because something wrong happened in a previous version.

Please inspect the content of the database and send me the data of an user that can't save his settings (hide sensitive info)
If you're unsure how to do this, hop onto Discord and I'll help you.

**Comment by pafade89** [comment]
Is this what you need?

![image](https://github.com/user-attachments/assets/7c3d4083-466b-40ab-ae32-45a52b4c63ea)
![image](https://github.com/user-attachments/assets/4fba218b-2e99-4e35-b0d7-6f5dcec0d09f)

**Comment by pafade89** [comment]
@gauthier-th I found the issue. When you delete an email address from a user profile, the database field is automatically filled with the username in lowercase letters. Required is Jellyfin user name written in the same way. If I do this manually in the db everything works fine. But when you re-enter the email address and delete it again the error is back.

**Comment by gauthier-th** [comment]
It look likes the email/usernames are mixed up, probably because of something that happened in a previous version.
I think the solution here would be to set a different email for every user, and then delete these emails.

Explanation:
The email field is mandatory because everything is based on that in Jellyseerr (this requirement is coming from Overseerr).
So what we did to make the email field not mandatory is using the Jellyfin username instead.
This means that if an user A has an email B, an user B without email will try to set it's email as B: 2 users would have the same email and this is not possible.

**Comment by pafade89** [comment]
@gauthier-th At first a Happy new Year =)
I have assigned all accounts with different email addresses, and when the emails are deleted, the error reoccurs. The email address is filled with the username but only in lowercase letters, which probably results in the comparison against the username not working.

**Comment by gauthier-th** [comment]
> Is this what you need?
> 
> ![image](https://private-user-images.githubusercontent.com/113902279/399311799-7c3d4083-466b-40ab-ae32-45a52b4c63ea.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MzU4MTU1OTUsIm5iZiI6MTczNTgxNTI5NSwicGF0aCI6Ii8xMTM5MDIyNzkvMzk5MzExNzk5LTdjM2Q0MDgzLTQ2NmItNDBhYi1hZTMyLTQ1YTUyYjRjNjNlYS5wbmc_WC1BbXotQWxnb3JpdGhtPUFXUzQtSE1BQy1TSEEyNTYmWC1BbXotQ3JlZGVudGlhbD1BS0lBVkNPRFlMU0E1M1BRSzRaQSUyRjIwMjUwMTAyJTJGdXMtZWFzdC0xJTJGczMlMkZhd3M0X3JlcXVlc3QmWC1BbXotRGF0ZT0yMDI1MDEwMlQxMDU0NTVaJlgtQW16LUV4cGlyZXM9MzAwJlgtQW16LVNpZ25hdHVyZT0xNjUzMTEyMDYwYzdlM2EwOWNmZmFhOGQyNjc4YmIzOTQwNWJiMjY1YWVmMTM3NDIzNzQzMDI1YzJjYjBmN2I0JlgtQW16LVNpZ25lZEhlYWRlcnM9aG9zdCJ9.SHrI-3CcFPeUPXkAYyj5qoH8rzMqaPQd1m43XQyqNvA) 

Why user `3` has `4` as username? And user `4` has `5` as username? I highly suspect it's because of this weird mix-up between all username and email.
And why every user has a number instead of an username?

As none of us can reproduce this, can you please try to setup a new instance with a fresh database to check if you can reproduce again the same issue?

**Comment by pafade89** [comment]
I had only manually changed the names once and unfortunately got one line wrong. However, the error is basically as described. I can also show it to you live on Discord if you'd like.

**Comment by gauthier-th** [comment]
> I had only manually changed the names once and unfortunately got one line wrong. However, the error is basically as described. I can also show it to you live on Discord if you'd like.

I'd prefer a screen recording of what's happening, and this:
> As none of us can reproduce this, can you please try to setup a new instance with a fresh database to check if you can reproduce again the same issue?

**Comment by pafade89** [comment]
I just created a completely new instance, and the error occurs immediately here as well.

Video per discord pm.

**Comment by pafade89** [comment]
Any news ?

**Comment by Fallenbagel** [comment]
> Any news ?

The only thing we can think of given none of us can reproduce this is, if it is possible for you to test on a fresh jellyfin instance as well. Both fresh jellyseer and jellyfin to eliminate any kind of user data corruption from jellyfin side

**Comment by pafade89** [comment]
@gauthier-th and @fallenbagel 
Thank you very much. Problem is now fixed #1389

**Comment by fallenbagel** [comment]
> @gauthier-th and @fallenbagel 
> Thank you very much. Problem is now fixed #1389 

FINALLY! IM SO GLAD 🎉

**Comment by gauthier-th** [comment]
> [@gauthier-th](https://github.com/gauthier-th) and [@fallenbagel](https://github.com/fallenbagel) Thank you very much. Problem is now fixed [#1389](https://github.com/fallenbagel/jellyseerr/pull/1389)

Glad it's finally working 😅
